TestQRdelete_F.test_delete_last_p_col test failure

My issue is about test failure for `scipy.linalg.tests.test_decomp_update.TestQRdelete_F`.
I sync-ed with master and updated cython to 0.29.13 (from 0.29.6) which it seems is newly required. I did `git clean -Xdf` and ran a full rebuild and test harness run before starting anything to make sure things are working. There was 1 test failure (output beneath). It looks like a numerical accuracy change occured somewhere in the last 80 commits or so. Either that or the cython update had an impact on accuracy, not sure the root cause.

The tip of my master has sha `commit 4d0d8958ad3f788a1a1c0bcac5cec1af9db26804`. 

#### Reproducing code example:

```
python runtests.py -v
```

#### Error message:
```
(scipy-dev) Lucass-MacBook:scipy rlucas$ python runtests.py -v
Building, see build.log...
    ... build in progress (0:01:00.273219 elapsed)
    ... build in progress (0:02:00.558959 elapsed)
    ... build in progress (0:03:00.799589 elapsed)
    ... build in progress (0:04:01.096821 elapsed)
    ... build in progress (0:05:01.418178 elapsed)
    ... build in progress (0:06:01.716540 elapsed)
    ... build in progress (0:07:02.059306 elapsed)
    ... build in progress (0:08:02.359432 elapsed)
    ... build in progress (0:09:02.673668 elapsed)
    ... build in progress (0:10:03.005183 elapsed)
Build OK (0:10:16.573951 elapsed)
================================================================================ test session starts ================================================================================
platform darwin -- Python 3.6.7, pytest-4.0.1, py-1.7.0, pluggy-0.8.0 -- /Users/rlucas/anaconda3/envs/scipy-dev/bin/python
cachedir: .pytest_cache
rootdir: /Users/rlucas/scipy-dev/scipy, inifile: pytest.ini
collected 20401 items / 1076 deselected                                                                                                                                             

scipy/_build_utils/tests/test_circular_imports.py::test_importing_submodules PASSED                                                                                           [  0%]
scipy/_build_utils/tests/test_scipy_version.py::test_valid_scipy_version PASSED                                                                                               [  0%]
scipy/_lib/tests/test__gcutils.py::test_set_gc_state PASSED                                                                                                                   [  0%]
scipy/_lib/tests/test__gcutils.py::test_gc_state PASSED                                                                                                                       [  0%]
scipy/_lib/tests/test__gcutils.py::test_assert_deallocated PASSED                                                                                                             [  0%]
scipy/_lib/tests/test__gcutils.py::test_assert_deallocated_nodel PASSED                                                                                                       [  0%]
scipy/_lib/tests/test__gcutils.py::test_assert_deallocated_circular PASSED                                                                                                    [  0%]
scipy/_lib/tests/test__gcutils.py::test_assert_deallocated_circular2 PASSED                                                                                                   [  0%]
scipy/_lib/tests/test__testutils.py::test__parse_size PASSED                                                                                                                  [  0%]
scipy/_lib/tests/test__testutils.py::test__mem_available PASSED                                                                                                               [  0%]
scipy/_lib/tests/test__threadsafety.py::test_parallel_threads PASSED        
#Note: eliding full test harness output for brevity
scipy/stats/tests/test_stats.py::TestRatioUniforms::test_exceptions PASSED                                                                                                    [ 99%]
scipy/stats/tests/test_stats.py::TestRatioUniforms::test_gig PASSED                                                                                                           [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_statistic_1 PASSED                                                                                                   [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_statistic_2 PASSED                                                                                                   [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_epps_singleton_array_like PASSED                                                                                     [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_epps_singleton_size PASSED                                                                                           [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_epps_singleton_nonfinite PASSED                                                                                      [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_epps_singleton_1d_input PASSED                                                                                       [ 99%]
scipy/stats/tests/test_stats.py::TestEppsSingleton::test_names PASSED                                                                                                         [ 99%]
scipy/stats/tests/test_tukeylambda_stats.py::test_tukeylambda_stats_known_exact PASSED                                                                                        [ 99%]
scipy/stats/tests/test_tukeylambda_stats.py::test_tukeylambda_stats_mpmath PASSED                                                                                             [ 99%]
scipy/stats/tests/test_tukeylambda_stats.py::test_tukeylambda_stats_invalid PASSED                                                                                            [100%]

===================================================================================== FAILURES ======================================================================================
_______________________________________________________________________ TestQRdelete_F.test_delete_last_p_col _______________________________________________________________________

self = <scipy.linalg.tests.test_decomp_update.TestQRdelete_F object at 0x11ea76208>

    def test_delete_last_p_col(self):
        a, q, r = self.generate('tall', 'full')
        q1, r1 = qr_delete(q, r, 0, a.shape[1], 'col')
>       assert_unitary(q1)

a          = array([[0.36852294+0.7065659j , 0.9763755 +0.8815727j ,
        0.73179656+0.5390478j , 0.6848747 +0.2577209j ,
      ...0.8069605j ,
        0.2923851 +0.9023696j , 0.08459105+0.7541054j ,
        0.8391204 +0.23152167j]], dtype=complex64)
q          = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
q1         = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
r          = array([[-3.1992414 +0.j        , -2.1899545 +0.23892963j,
        -1.6467261 +0.47293147j, -2.4033258 +0.41776413j,
  ...        ,
         0.        +0.j        ,  0.        +0.j        ,
         0.        +0.j        ]], dtype=complex64)
r1         = array([], shape=(12, 0), dtype=complex64)
self       = <scipy.linalg.tests.test_decomp_update.TestQRdelete_F object at 0x11ea76208>

scipy/linalg/tests/test_decomp_update.py:327: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
rtol = 0.0001, atol = 2.384185791015625e-07, assert_sqr = True

    def assert_unitary(a, rtol=None, atol=None, assert_sqr=True):
        if rtol is None:
            rtol = 10.0 ** -(np.finfo(a.dtype).precision-2)
        if atol is None:
            atol = 2*np.finfo(a.dtype).eps
    
        if assert_sqr:
            assert_(a.shape[0] == a.shape[1], 'unitary matrices must be square')
        aTa = np.dot(a.T.conj(), a)
>       assert_allclose(aTa, np.eye(a.shape[1]), rtol=rtol, atol=atol)
E       AssertionError: 
E       Not equal to tolerance rtol=0.0001, atol=2.38419e-07
E       
E       Mismatch: 1.39%
E       Max absolute difference: 3.57627869e-07
E       Max relative difference: inf
E        x: array([[ 1.000000e+00+0.000000e+00j, -1.676381e-08+1.490116e-08j,
E                0.000000e+00+1.490116e-08j, -2.793968e-08-2.980232e-08j,
E               -2.235174e-08-1.862645e-08j, -2.095476e-08+5.587935e-09j,...
E        y: array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
E              [0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
E              [0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.],...

a          = array([[-0.11519074-0.22085421j,  0.4031076 +0.25545135j,
         0.20641464-0.02725353j,  0.04916074+0.12054591j,
  ....17945503j,  0.10257003-0.11285198j,
        -0.05898733+0.03267026j,  0.5694973 -0.06679238j]],
      dtype=complex64)
aTa        = array([[ 1.0000001e+00+0.0000000e+00j, -1.6763806e-08+1.4901161e-08j,
         0.0000000e+00+1.4901161e-08j, -2.793967...3484e-08-4.4703484e-08j,
        -8.3819032e-09-2.9802322e-08j,  9.9999994e-01+0.0000000e+00j]],
      dtype=complex64)
assert_sqr = True
atol       = 2.384185791015625e-07
rtol       = 0.0001

scipy/linalg/tests/test_decomp_update.py:21: AssertionError
========================================== 1 failed, 17950 passed, 1291 skipped, 1076 deselected, 76 xfailed, 7 xpassed in 530.96 seconds ===========================================

   Normal return from subroutine COBYLA

   NFVALS =   50   F = 2.485185E+01    MAXCV = 1.999965E-10
   X = 4.955358E+00   6.666553E-01

 NNLS quitting on iteration count.
(scipy-dev) Lucass-MacBook:scipy rlucas$
```

#### Scipy/Numpy/Python version information:
```
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.4.0.dev0+4d0d895 1.16.4 sys.version_info(major=3, minor=6, micro=7, releaselevel='final', serial=0)
```

--- a/scipy/linalg/tests/test_decomp_update.py
+++ b/scipy/linalg/tests/test_decomp_update.py
@@ -7,13 +7,13 @@ from numpy.testing import assert_, assert_allclose, assert_equal
 from pytest import raises as assert_raises
 from scipy import linalg
 import scipy.linalg._decomp_update as _decomp_update
-from scipy.linalg._decomp_update import *
+from scipy.linalg._decomp_update import qr_delete, qr_update, qr_insert
 
 def assert_unitary(a, rtol=None, atol=None, assert_sqr=True):
     if rtol is None:
         rtol = 10.0 ** -(np.finfo(a.dtype).precision-2)
     if atol is None:
-        atol = 2*np.finfo(a.dtype).eps
+        atol = 10*np.finfo(a.dtype).eps
 
     if assert_sqr:
         assert_(a.shape[0] == a.shape[1], 'unitary matrices must be square')
@@ -49,11 +49,13 @@ def make_strided(arrs):
             s = strides[k % kmax]
             t = strides[(k+1) % kmax]
             k += 2
-            base = np.zeros((s[0]*a.shape[0]+s[1], t[0]*a.shape[1]+t[1]), a.dtype)
+            base = np.zeros((s[0]*a.shape[0]+s[1], t[0]*a.shape[1]+t[1]),
+                            a.dtype)
             view = base[s[1]::s[0], t[1]::t[0]]
             view[...] = a
         else:
-            raise ValueError('make_strided only works for ndim = 1 or 2 arrays')
+            raise ValueError('make_strided only works for ndim = 1 or'
+                             ' 2 arrays')
         ret.append(view)
     return ret
 
@@ -66,7 +68,8 @@ def negate_strides(arrs):
         elif b.ndim == 1:
             b = b[::-1]
         else:
-            raise ValueError('negate_strides only works for ndim = 1 or 2 arrays')
+            raise ValueError('negate_strides only works for ndim = 1 or'
+                             ' 2 arrays')
         b[...] = a
         ret.append(b)
     return ret
@@ -205,7 +208,7 @@ class BaseQRdelete(BaseQRdeltas):
             a1 = np.delete(a, row, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol, False)
 
-    # for economic row deletes 
+    # for economic row deletes
     # eco - prow = eco
     # eco - prow = sqr
     # eco - prow = fat
@@ -291,7 +294,7 @@ class BaseQRdelete(BaseQRdeltas):
                 check_qr(q1, r1, a1, self.rtol, self.atol, False)
 
     def test_delete_last_1_row(self):
-        # full and eco are the same for 1xN 
+        # full and eco are the same for 1xN
         a, q, r = self.generate('1xN')
         q1, r1 = qr_delete(q, r, 0, 1, 'row')
         assert_equal(q1, np.ndarray(shape=(0, 0), dtype=q.dtype))
@@ -350,10 +353,11 @@ class BaseQRdelete(BaseQRdeltas):
     # all full qr, row deletes and single column deletes should be able to
     # handle any non negative strides. (only row and column vector
     # operations are used.) p column delete require fortran ordered
-    # Q and R and will make a copy as necessary.  Economic qr row deletes 
+    # Q and R and will make a copy as necessary.  Economic qr row deletes
     # requre a contigous q.
 
-    def base_non_simple_strides(self, adjust_strides, ks, p, which, overwriteable):
+    def base_non_simple_strides(self, adjust_strides, ks, p, which,
+                                overwriteable):
         if which == 'row':
             qind = (slice(p,None), slice(p,None))
             rind = (slice(p,None), slice(None))
@@ -369,7 +373,8 @@ class BaseQRdelete(BaseQRdeltas):
             else:
                 s = slice(k,k+p)
                 if k < 0:
-                    s = slice(k, k + p + (a.shape[0] if which == 'row' else a.shape[1]))
+                    s = slice(k, k + p +
+                              (a.shape[0] if which == 'row' else a.shape[1]))
                 a1 = np.delete(a, s, 0 if which == 'row' else 1)
 
             # for each variable, q, r we try with it strided and
@@ -377,7 +382,7 @@ class BaseQRdelete(BaseQRdeltas):
             # sure that q and r are still overwritten.
 
             q = q0.copy('F')
-            r = r0.copy('F') 
+            r = r0.copy('F')
             q1, r1 = qr_delete(qs, r, k, p, which, False)
             check_qr(q1, r1, a1, self.rtol, self.atol)
             q1o, r1o = qr_delete(qs, r, k, p, which, True)
@@ -527,7 +532,7 @@ class BaseQRdelete(BaseQRdeltas):
 
     def test_overwrite_qr_p_col(self):
         # only F orderd q and r can be overwritten for cols
-        # full and eco share code paths 
+        # full and eco share code paths
         self.base_overwrite_qr('col', 3, False, True)
 
     def test_bad_which(self):
@@ -571,7 +576,7 @@ class BaseQRdelete(BaseQRdeltas):
         assert_raises(ValueError, qr_delete, q, r, 0, 1)
 
     def test_unsupported_dtypes(self):
-        dts = ['int8', 'int16', 'int32', 'int64', 
+        dts = ['int8', 'int16', 'int32', 'int64',
                'uint8', 'uint16', 'uint32', 'uint64',
                'float16', 'longdouble', 'longcomplex',
                'bool']
@@ -658,7 +663,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, row)
             a1 = np.insert(a, row, u, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol)
-    
+
     def test_sqr_p_row(self):
         # sqr + rows --> fat always
         a, q, r, u = self.generate('sqr', which='row', p=3)
@@ -744,7 +749,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, row)
             a1 = np.insert(a, np.full(p, row, np.intp), u, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol)
-    
+
     def test_fat_p_row_fat(self):
         # 7x12 + 3x12 = 10x12 --> stays fat
         self.base_fat_p_row_xxx(3)
@@ -831,7 +836,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, row)
             a1 = np.insert(a, row, u, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol)
-    
+
     def test_Mx1_p_row(self):
         a, q, r, u = self.generate('Mx1', which='row', p=3)
         for row in range(r.shape[0] + 1):
@@ -859,7 +864,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, row)
             a1 = np.insert(a, row, u, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol, False)
-    
+
     def test_Mx1_economic_p_row(self):
         a, q, r, u = self.generate('Mx1', 'economic', 'row', 3)
         for row in range(r.shape[0] + 1):
@@ -887,7 +892,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, row)
             a1 = np.insert(a, row, u, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol)
-    
+
     def test_1xN_p_row(self):
         a, q, r, u = self.generate('1xN', which='row', p=3)
         for row in range(r.shape[0] + 1):
@@ -915,7 +920,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, row)
             a1 = np.insert(a, row, u, 0)
             check_qr(q1, r1, a1, self.rtol, self.atol)
-    
+
     def test_1x1_p_row(self):
         a, q, r, u = self.generate('1x1', which='row', p=3)
         for row in range(r.shape[0] + 1):
@@ -936,7 +941,7 @@ class BaseQRinsert(BaseQRdeltas):
             q1, r1 = qr_insert(q, r, u, col, 'col', overwrite_qru=False)
             a1 = np.insert(a, np.full(3, col, np.intp), u, 1)
             check_qr(q1, r1, a1, self.rtol, self.atol)
-    
+
     def test_1x1_1_scalar(self):
         a, q, r, u = self.generate('1x1', which='row')
         assert_raises(ValueError, qr_insert, q[0, 0], r, u, 0, 'row')
@@ -955,7 +960,7 @@ class BaseQRinsert(BaseQRdeltas):
                 ai = np.insert(a, k, u0, 0 if which == 'row' else 1)
             else:
                 ai = np.insert(a, np.full(p, k, np.intp),
-                        u0 if which == 'row' else u0, 
+                        u0 if which == 'row' else u0,
                         0 if which == 'row' else 1)
 
             # for each variable, q, r, u we try with it strided and
@@ -996,7 +1001,7 @@ class BaseQRinsert(BaseQRdeltas):
             check_qr(q5, r5, ai, self.rtol, self.atol)
             q5o, r5o = qr_insert(qs, rs, us, k, which, overwrite_qru=True)
             check_qr(q5o, r5o, ai, self.rtol, self.atol)
-    
+
     def test_non_unit_strides_1_row(self):
         self.base_non_simple_strides(make_strided, 0, 1, 'row')
 
@@ -1047,7 +1052,7 @@ class BaseQRinsert(BaseQRdeltas):
 
     def test_overwrite_qu_rank_1(self):
         # when inserting rows, the size of both Q and R change, so only
-        # column inserts can overwrite q. Only complex column inserts 
+        # column inserts can overwrite q. Only complex column inserts
         # with C ordered Q overwrite u. Any contiguous Q is overwritten
         # when inserting 1 column
         a, q0, r, u, = self.generate('sqr', which='col', p=1)
@@ -1115,7 +1120,7 @@ class BaseQRinsert(BaseQRdeltas):
         assert_raises(ValueError, qr_insert, q, r, u[1:], 0, 'col')
 
     def test_unsupported_dtypes(self):
-        dts = ['int8', 'int16', 'int32', 'int64', 
+        dts = ['int8', 'int16', 'int32', 'int64',
                'uint8', 'uint16', 'uint32', 'uint64',
                'float16', 'longdouble', 'longcomplex',
                'bool']
@@ -1262,7 +1267,7 @@ class BaseQRupdate(BaseQRdeltas):
         check_qr(q1, r1, a1, self.rtol, self.atol)
 
     def test_Mx1_rank_p(self):
-        # when M or N == 1, only a rank 1 update is allowed. This isn't 
+        # when M or N == 1, only a rank 1 update is allowed. This isn't
         # fundamental limitation, but the code does not support it.
         a, q, r, u, v = self.generate('Mx1', p=1)
         u = u.reshape(u.size, 1)
@@ -1278,7 +1283,7 @@ class BaseQRupdate(BaseQRdeltas):
         check_qr(q1, r1, a1, self.rtol, self.atol, False)
 
     def test_Mx1_economic_rank_p(self):
-        # when M or N == 1, only a rank 1 update is allowed. This isn't 
+        # when M or N == 1, only a rank 1 update is allowed. This isn't
         # fundamental limitation, but the code does not support it.
         a, q, r, u, v = self.generate('Mx1', 'economic', p=1)
         u = u.reshape(u.size, 1)
@@ -1294,7 +1299,7 @@ class BaseQRupdate(BaseQRdeltas):
         check_qr(q1, r1, a1, self.rtol, self.atol)
 
     def test_1xN_rank_p(self):
-        # when M or N == 1, only a rank 1 update is allowed. This isn't 
+        # when M or N == 1, only a rank 1 update is allowed. This isn't
         # fundamental limitation, but the code does not support it.
         a, q, r, u, v = self.generate('1xN', p=1)
         u = u.reshape(u.size, 1)
@@ -1310,7 +1315,7 @@ class BaseQRupdate(BaseQRdeltas):
         check_qr(q1, r1, a1, self.rtol, self.atol)
 
     def test_1x1_rank_p(self):
-        # when M or N == 1, only a rank 1 update is allowed. This isn't 
+        # when M or N == 1, only a rank 1 update is allowed. This isn't
         # fundamental limitation, but the code does not support it.
         a, q, r, u, v = self.generate('1x1', p=1)
         u = u.reshape(u.size, 1)
@@ -1426,7 +1431,7 @@ class BaseQRupdate(BaseQRdeltas):
 
     def test_neg_strides_economic_rank_p(self):
         self.base_non_simple_strides(negate_strides, 'economic', 3, False)
-     
+
     def test_non_itemsize_strides_rank_1(self):
         self.base_non_simple_strides(nonitemsize_strides, 'full', 1, False)
 
@@ -1452,7 +1457,7 @@ class BaseQRupdate(BaseQRdeltas):
         self.base_non_simple_strides(make_nonnative, 'economic', 3, False)
 
     def test_overwrite_qruv_rank_1(self):
-        # Any positive strided q, r, u, and v can be overwritten for a rank 1 
+        # Any positive strided q, r, u, and v can be overwritten for a rank 1
         # update, only checking C and F contiguous.
         a, q0, r0, u0, v0 = self.generate('sqr')
         a1 = a + np.outer(u0, v0.conj())
@@ -1548,7 +1553,7 @@ class BaseQRupdate(BaseQRdeltas):
         assert_raises(ValueError, qr_update, q, r, u, v[1:])
 
     def test_unsupported_dtypes(self):
-        dts = ['int8', 'int16', 'int32', 'int64', 
+        dts = ['int8', 'int16', 'int32', 'int64',
                'uint8', 'uint16', 'uint32', 'uint64',
                'float16', 'longdouble', 'longcomplex',
                'bool']
@@ -1642,7 +1647,7 @@ def test_form_qTu():
     # tested. Most of them should be hit with the rest of test suite, but
     # explicit tests make clear precisely what is being tested.
     #
-    # This function expects that Q is either C or F contiguous and square. 
+    # This function expects that Q is either C or F contiguous and square.
     # Economic mode decompositions (Q is (M, N), M != N) do not go through this
     # function. U may have any positive strides.
     #
@@ -1662,7 +1667,7 @@ def test_form_qTu():
             check_form_qTu(qo, qs, uo, us, 2, d)
         else:
             check_form_qTu(qo, qs, uo, us, 2, d)
-    
+
 def check_form_qTu(q_order, q_shape, u_order, u_shape, u_ndim, dtype):
     np.random.seed(47)
     if u_shape == 1 and u_ndim == 1:
@@ -1685,12 +1690,10 @@ def check_form_qTu(q_order, q_shape, u_order, u_shape, u_ndim, dtype):
         u = np.require(u, dtype, u_order)
     else:
         u, = make_strided((u.astype(dtype),))
-    
+
     rtol = 10.0 ** -(np.finfo(dtype).precision-2)
     atol = 2*np.finfo(dtype).eps
 
     expected = np.dot(q.T.conj(), u)
     res = _decomp_update._form_qTu(q, u)
     assert_allclose(res, expected, rtol=rtol, atol=atol)
-
- 

